pubsub/subscriptions/sync_pull.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscriptions
+
+// [START pubsub_subscriber_sync_pull]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/pubsub"
+	pubsubV1 "cloud.google.com/go/pubsub/apiv1"
+	pb "google.golang.org/genproto/googleapis/pubsub/v1"
+)
+
+func pullMsgsSync(w io.Writer, projectID, subName string, topic *pubsub.Topic, maxMessages int32) ([]string, error) {
+	// projectID := "my-project-id"
+	// subName := projectID + "-example-sub"
+	// topic of type https://godoc.org/cloud.google.com/go/pubsub#Topic
+	ctx := context.Background()
+
+	// Publish 10 messages on the topic.
+	var results []*pubsub.PublishResult
+	for i := 0; i < 15; i++ {
+		res := topic.Publish(ctx, &pubsub.Message{
+			Data: []byte(fmt.Sprintf("hello world #%d", i)),
+		})
+		results = append(results, res)
+	}
+
+	// Check that all messages were published.
+	for _, r := range results {
+		_, err := r.Get(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("Get: %v", err)
+		}
+	}
+
+	// Instantiate a client
+	subClient, err := pubsubV1.NewSubscriberClient(ctx)
```

Comment:
Should be using `cloud.google.com/go/pubsub` rather than one of the versioned packages.

Reply:
Not sure I fully understand. Sounds good to me, but adding @jadekler for additional review.

Reply:
@jadekler Is the comment clear enough? If you don't think so, we can remove all uses of the gapic client altogether and just focus on the functionality exposed in the handwritten client.

Reply:
This method is useful when only a specific number of messages are wanted by an end application that takes a long time to process each incoming message. Because Pub/Sub can flood the end application with tons of messages, many messages would make wasted trips across the network if we don't avoid asking for them. 

@hongalex [The docs site](https://cloud.google.com/pubsub/docs/pull#synchronous_pull) mentions that this is a REST method, not gRPC, is that correct? If not, we may still want to call out that this is a [`PullRequest`](https://cloud.google.com/pubsub/docs/reference/rpc/google.pubsub.v1#google.pubsub.v1.PullRequest) is not [`StreamingPullRequest`](https://cloud.google.com/pubsub/docs/reference/rpc/google.pubsub.v1#streamingpullrequest).